sdk/core/src/main/java/com/klaviyo/coresdk/model/Profile.kt

```diff
@@ -0,0 +1,92 @@
+package com.klaviyo.coresdk.model
+
+import java.io.Serializable
+
+/**
+ * Controls the data that can be input into a map of profile attributes recognised by Klaviyo
+ */
+class Profile(properties: Map<ProfileKey, Serializable>?) :
+    BaseModel<ProfileKey, Profile>(properties) {
+
+    constructor() : this(null)
+
+    private var appendMap: HashMap<String, Serializable> = HashMap()
+
+    fun setExternalId(identifier: String) = apply { this.externalId = identifier }
+    var externalId: String?
+        get() = (this[ProfileKey.EXTERNAL_ID]) as String?
+        set(value) {
+            this[ProfileKey.EXTERNAL_ID] = value
+        }
+
+    fun setEmail(email: String) = apply { this.email = email }
+    var email: String?
+        get() = (this[ProfileKey.EMAIL]) as String?
+        set(value) {
+            this[ProfileKey.EMAIL] = value
+        }
+
+    fun setPhoneNumber(phoneNumber: String) = apply { this.phoneNumber = phoneNumber }
+    var phoneNumber: String?
+        get() = (this[ProfileKey.PHONE_NUMBER]) as String?
+        set(value) {
+            this[ProfileKey.PHONE_NUMBER] = value
+        }
+
+    internal fun setAnonymousId(anonymousId: String) = apply { this.anonymousId = anonymousId }
+    internal var anonymousId: String?
+        get() = (this[ProfileKey.ANONYMOUS]) as String
+        set(value) {
+            this[ProfileKey.ANONYMOUS] = value
+        }
+
+    // TODO internal?
```

Comment:
What are the cons of leaving this exposed?

Reply:
Just a potential confusion, it really shouldn't be used for anything but a push token. I've actually taken this out while working on the V3 profile API though because V3 api doesn't even recognize `$append`.

Reply:
Cool ok, so sounds inconsequential to this PR!

Reply:
Yup, PR for V3 coming shortly and I think you'll see what I mean. I just put the handling of `$append` straight into the push token API Request class.